sdk/communication/Azure.Communication.CallAutomation/src/CallDialog.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.CallAutomation
+{
+    /// <summary>
+    /// The Azure Communication Services Call Dialog Client.
+    /// </summary>
+    public class CallDialog
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+        internal CallDialogRestClient CallDialogRestClient { get; }
+        internal CallAutomationEventProcessor EventProcessor { get; }
+
+        /// <summary>
+        /// The call connection id.
+        /// </summary>
+        public virtual string CallConnectionId { get; internal set; }
+
+        internal CallDialog(string callConnectionId, CallDialogRestClient callDialogRestClient, ClientDiagnostics clientDiagnostics, CallAutomationEventProcessor eventProcessor)
+        {
+            _clientDiagnostics = clientDiagnostics;
+            CallDialogRestClient = callDialogRestClient;
+            EventProcessor = eventProcessor;
+            CallConnectionId = callConnectionId;
+        }
+
+        /// <summary>Initializes a new instance of <see cref="CallDialog"/> for mocking.</summary>
+        protected CallDialog()
+        {
+            _clientDiagnostics = null;
+            CallDialogRestClient = null;
+            CallConnectionId = null;
+        }
+        /// <summary>
+        /// Start Dialog.
+        /// </summary>
+        /// <param name="startDialogOptions">Configuration attributes for starting dialog.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>Returns <see cref="DialogResult"/>, which can be used to wait for Dialog's related events.</returns>
+        public virtual async Task<Response<DialogResult>> StartDialogAsync(StartDialogOptions startDialogOptions, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallDialog)}.{nameof(StartDialog)}");
+            scope.Start();
+            try
+            {
+                StartDialogRequestInternal request = CreateStartDialogRequest(startDialogOptions);
+
+                var response = await CallDialogRestClient.StartDialogAsync
+                    (CallConnectionId,
+                    startDialogOptions.DialogId,
+                    request,
+                    cancellationToken).ConfigureAwait(false);
+
+                var result = new DialogResult(startDialogOptions.DialogId);
+                result.SetEventProcessor(EventProcessor, CallConnectionId, request.OperationContext);
```

Comment:
when utilizing event processor, make sure operation context is set if empty.

> request.OperationContext = options.OperationContext == default ? Guid.NewGuid().ToString() : options.OperationContext;

Reply:
I'll update all the calls within dialog - should the ones in CallMedia also be updated accordingly?

Reply:
Added this to the CreateStartDialogRequest method